ocaml/xenopsd/c_stubs/tuntap_stubs.c

```diff
@@ -41,20 +44,23 @@ CAMLprim value stub_tap_open(value ocaml_ifname)
     }
     strncpy(ifr.ifr_name, ifname, IFNAMSIZ);
 
+    path_net_tun = caml_copy_string(PATH_NET_TUN);
     int fd = open(PATH_NET_TUN, O_RDWR);
     if (fd < 0) {
-        caml_failwith("open(" PATH_NET_TUN ") failed in " __FILE__);
+        uerror("open", path_net_tun);
```

Comment:
If you want to preserve the file, you could use string concatenation of static strings in GCC: `"open failed in " __FILE__` Is a single string.

Reply:
According to https://v2.ocaml.org/api/Unix.html the string there should be the function name,
I could do something like `"open at " __FILE__`, which would print something like this when converting the exception to string:
`"open at ..." failed on "...file...": error description`

Although in this case '/dev/net/tun' is a unique string in the code, and I hope the stacktrace will also provide enough context where the error is raised at. Previously the filename was not printed at all so __FILE__ was essential with locating where this comes from, for now lets try to follow the usual conventions of Unix.error.